<?xml version="1.0"?>
<launch>
    <arg name="realsense" value="true"/>
    <arg name="camera_ns" value="/camera" />

    <group if="$(arg realsense)">
        <include file="$(find arm_vs)/launch/process_pointcloud.launch">
            <arg name="pointcloud_input_topic"  value="/point_cloud/cloud_registered"/>
            <arg name="openni_device"           value="false"/>
        </include>
        
        <include file="$(find arm_vs)/launch/obj_image_tracking.launch">
            <arg name="tracker_type"            value="GOTURN" />
            <arg name="tracking_image_source"   value="/rgb/image_raw_color" />
            <arg name="pointcloud_source"       value="/voxel_grid/output" />
        </include>
    </group>

    <group unless="$(arg realsense)">
        <include file="$(find arm_vs)/launch/process_pointcloud.launch">
            <arg name="pointcloud_input_topic"  value="/camera/depth/points"/>
            <arg name="openni_device"           value="true"/>
        </include>
        
        <include file="$(find arm_vs)/launch/obj_image_tracking.launch">
            <arg name="tracker_type"            value="GOTURN" />
            <arg name="tracking_image_source"   value="$(arg camera_ns)/rgb/image_raw" />
            <arg name="pointcloud_source"       value="$(arg camera_ns)/depth/points" />
        </include>
    </group>

</launch>